hibernate-core/src/main/java/org/hibernate/query/results/DomainResultCreationStateImpl.java

```diff
@@ -266,7 +268,8 @@ public Expression resolveSqlExpression(
 		else if ( created instanceof ColumnReference ) {
 			final ColumnReference columnReference = (ColumnReference) created;
 			final String columnExpression = columnReference.getColumnExpression();
-			final int jdbcPosition = jdbcResultsMetadata.resolveColumnPosition( columnExpression );
+
+			final int jdbcPosition = jdbcResultsMetadata.resolveColumnPosition( currentResultBuilder.getColumnAlias( columnExpression ) );
```

Comment:
This and all the related changes shouldn't be necessary. What does that fix?

Reply:
reverted thanks